File: web/src/api/pvc.ts
import service from '@/utils/request';

// 获取PVC列表
export const getPVCList = (params?: any) => {
    return service({
        url: '/api/v1/pvc/list',
        method: 'get',
        params
    });
};

// 创建PVC
export const createPVC = (data: any) => {
    return service({
        url: '/api/v1/pvc/add',
        method: 'post',
        data
    });
};

// 删除PVC
export const deletePVC = (data: any) => {
    return service({
        url: '/api/v1/pvc/delete',
        method: 'post',
        data
    });
};

// 更新PVC
export const updatePVC = (data: any) => {
    return service({
        url: '/api/v1/pvc/update',
        method: 'post',
        data
    });
};

File: server/router/pvc/pvc.go
package pvc

import (
	"gin-vue-admin/middleware"

	"github.com/gin-gonic/gin"
)

type PVCRouter struct{}

func (s *PVCRouter) InitPVCRouter(PrivateGroup *gin.RouterGroup) {
	apiRouter := PrivateGroup.Group("pvc").Use(middleware.OperationRecord())
	apiWithNotRecorderRouter := PrivateGroup.Group("pvc")
	{
		apiRouter.POST("add", pvcApi.AddPVC)       // 创建PVC
		apiRouter.POST("delete", pvcApi.DeletePVC) // 删除PVC
		apiRouter.POST("update", pvcApi.UpdatePVC) // 更新PVC
	}
	{
		apiWithNotRecorderRouter.GET("list", pvcApi.GetPVCList) // 获取PVC列表
	}
}
